userspace/libsinsp/CMakeLists.txt

```diff
@@ -169,12 +169,14 @@ target_link_libraries(sinsp
 			"${RE2_LIB}"
 )
 
-set(SINSP_PKGCONFIG_LIBRARIES
-	scap
-	"${ZLIB_LIB}"
-	"${CURL_LIBRARIES}"
-	"${JSONCPP_LIB}"
-	"${RE2_LIB}"
+set(SINSP_PKGCONFIG_LIBRARIES)
+set(SINSP_PKGCONFIG_REQUIRES
+	$<$<BOOL:${JSONCPP_LIB}>:jsoncpp>
+)
+
+set(SINSP_PKGCONFIG_REQUIRES_PRIVATE
+	$<$<BOOL:${CURL_LIBRARIES}>:libcurl>
+	$<$<BOOL:${RE2_LIB}>:re2>
```

Comment:
Ouch. Why this?
I know we don't always use `find_package` which would provide nice target aliases like `CURL::libcurl` and `re2::re2`, which bring on board all the properties a "consumer" of those libraries would need (include path, binary to link to, etc.). But why this? It seems like a workaround to give a target when the variable is not available. I'd address the issue upstream and ensure the variable has a value. Still inferior to having the standard target aliases for each library, but not relying on generator expression black magic. Please, forgive me for being direct on this. I do believe simplicity is key when dealing with CMake.

Reply:
It was a defensive technique to avoid adding a pkg-config module name in case a library was not found and this was not a fatal error.

Looking at it more closely, it seems this should never occur, so I've removed these safeguards, for better readability.